### PR TITLE
chore: Remove deprecated eslint rules in favor of stylistic rules

### PR DESCRIPTION
**Pull request by marikaner**
Some rules are deprecated (e.g. https://typescript-eslint.io/rules/member-delimiter-style/) and are not replaced by ESLint Stylistic. Those rules are removed in later versions of typescript eslint, so I removed them before the dependabot PR (there are other issues as well).

### CODE REVIEW

File: packages/eslint-config/package.json [hunk 1]
```diff
@@ -25,16 +25,20 @@
...
     "eslint-plugin-unused-imports": "^4.1.4",
-    "typescript-eslint": "^7.18.0",
-    "@eslint/js": "^9.11.1"
+    "typescript-eslint": "^7.18.0"
   },
   "peerDependencies": {
     "eslint": "^8.0.0"
+  },
+  "devDependencies": {
+    "@stylistic/eslint-plugin-migrate": "^2.8.0"
```
Comment (deekshas8) [inline packages/eslint-config/package.json:42]: [q] Is this needed? Briefly looking [here](https://eslint.style/guide/migration#eslint-migrate-plugin) it says we need to add comments at the top of the config file to automatically migrate rules.

File: test-resources/test/test-util/xsuaa-service-mocks.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export function mockRefreshTokenGrantCall(
 
 function xsuaaRequestHeaders(additionalHeaders: Record<string, string> = {}) {
   return {
-    Accept: 'application/json',
+    accept: 'application/json',
```
Comment (deekshas8) [inline test-resources/test/test-util/xsuaa-service-mocks.ts:90]: Scope creep ;)

File: packages/eslint-config/package.json (hunk 1 above)
Comment (marikaner) [reply packages/eslint-config/package.json]: no not needed anymore, also just realized this and removed it 😊

**Review by deekshas8 — APPROVED** [review]
LGTM. Left one question.

File: test-resources/test/test-util/xsuaa-service-mocks.ts (hunk 2 above)
Comment (marikaner) [reply test-resources/test/test-util/xsuaa-service-mocks.ts:90]: yes 🫣
